src/transformers/pipelines/conversational.py

```diff
@@ -163,7 +165,13 @@ def _user_messages(self):
     @property
     def past_user_inputs(self):
         # This is a legacy property for backwards compatibility. It is recommended to just directly access
-        # conversation.messages instead.
+        # conversation.messages instead. The modern class does not care about which messages are "processed"
+        # or not.
+        if not self._user_messages:
+            return []
+        if self.messages[-1]["role"] != "user" or self._num_processed_user_inputs == len(self._user_messages):
+            return self._user_messages
+
         return self._user_messages[:-1]
```

Comment:
Could we have a bit of comments about why this conditions and their output values?

Reply:
I mean hard to understand why `self._user_messages` when `self.messages[-1]["role"] != "user"`

Reply:
Added a comment!